lib/fs/fakefs.go

```diff
@@ -444,13 +499,15 @@ func (fs *fakefs) Rename(oldname, newname string) error {
 		return os.ErrNotExist
 	}
 
-	dst, ok := p1.children[filepath.Base(newname)]
+	dst, ok := p1.children[newKey]
 	if ok && dst.entryType == fakeEntryTypeDir {
```

Comment:
I think this results in error `is a directory` when doing a case-only rename of an existing directory?

Reply:
Yep, wrote a failing test, expect commits later today.

Reply:
Actually, here you go, done.